### ISSUE TITLE
No slug generation and validation errors for not currently selected sites when saving drafts

### ISSUE THREAD

**Issue by wsydney76** [craftcms/cms#5097]
### Description

It looks like when saving a draft slug generation and validations are only performed for the entry on the currently selected site, but not for others.

### Steps to reproduce

1. Setup: Two sites (en/de), a section with a required translatable field
2. Create a new entry in en, fill out title and the field => a slug is generated
3. Switch to de
4. Delete content in slug and the required field, fill out a german title, => slug stays empty
5. Switch to en
6. Click save => entry gets saved without errors.

There is no validation error for the empty field in de, the slug is still empty, resulting in a uri like /__temp_MGARX4UVx4PPP6Sj0PjK6o8Z3dze0cYSKMjU

(Save after step 4 and everything works fine.)

Thanks.

### Additional info

- Craft version: 3.3.9
- PHP version: 7.2.14
- Database driver & version: MySQL 5.7.14
- Plugins & versions: none

**Comment by brandonkelly** [comment]
Thanks for reporting that! It’s been fixed for the next release.

**Comment by wsydney76** [comment]
@brandonkelly Thanks, but after updating to 3.3.10 we see some issues, at first glance caused by line 782 in  services/Elements, which seems to be invoked if the duplicating element is a matrix block.

* When trying to create a new draft, the call returns with `Element 1209 could not be duplicated for site 2: Slug cannot be blank.` where 1209 is a matrix block.
* When trying to save a draft there is

![image](https://user-images.githubusercontent.com/25308802/66865015-ae4e2300-ef96-11e9-9df9-39d2402b2fb6.png)

**Comment by brandonkelly** [comment]
@wsydney76 I’m not able to reproduce that. Can you send your `composer.json` and `composer.lock` files, plus a database backup, over to support@craftcms.com?

**Comment by brandonkelly** [comment]
We just released Craft 3.3.11 with a fix for this.

**Comment by wsydney76** [comment]
@brandonkelly Thanks, the slug generation works fine now. However the fix does not seem to target at the second problem (missing validations for not active sites). May be i should open a new issue for that?

**Comment by brandonkelly** [comment]
Yeah new issue would be great.
